main.tf

```diff
@@ -50,13 +54,12 @@ resource "azurerm_linux_web_app" "main" {
     "SECRET_KEY"                 = var.secret_key
```

Comment:
I know we saw issues trying to use the generated Mongo connection string so happy that you stuck with the variable here, but I managed to track down that resolving the deprecation warning seemed to resolve the permissions issue too for someone else, so if you're interested we could try switching out the `var.mongodb_connection_string` for `azurerm_cosmosdb_account.terraCosmos.primary_mongodb_connection_string`